components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -236,4 +340,182 @@ private String getProcessStreamId() {
+                            + ", Process Definition Id:" + processDefinitionId + ", Published Event's Payload Data :"
+                            + payload.toString());
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Failed Publishing BPMN process instance KPI event... Process Instance Id :" +
+                            processInstanceId + ", Process Definition Id:" + processDefinitionId
+                            + ", Published Event's Payload Data :" + payload.toString());
+                }
+            }
+        } catch (RegistryException | RuntimeException e) {
+            String strMsg = "Failed Publishing BPMN process instance KPI event... Process Instance Id :" +
+                    processInstanceId + ", Process Definition Id:" + processDefinitionId
+                    + ", Published Event's Payload Data :" + payload.toString();
+            throw new BPMNDataPublisherException(strMsg, e);
+        }
+    }
+
+    /**
+     * Get DAS config details of given certain process which are configured for analytics from the config registry
+     *
+     * @param processDefinitionId
+     * @return KPI configuration details in JSON format. Ex:<p>
+     * {"eventReceiverName":"process1_77_process_receiver",
+     * "processDefinitionId":"manualTaskProcess111:1:22509","pcProcessId":"process1:77",
+     * "eventStreamNickName":"process1_77_process_stream","eventStreamDescription":"This is the event stream
+     * generated to configure process analytics with DAS, for the processprocess1_77","eventStreamVersion":"1.0.0",
+     * "eventStreamId":"process1_77_process_stream:1.0.0","processVariables":[{"isAnalyzeData":false,"name":"size",
+     * "isDrillDownData":false,"type":"int"},{"isAnalyzeData":false,"name":"pizzaTopping","isDrillDownData":false,
+     * "type":"string"},{"isAnalyzeData":false,"name":"amount","isDrillDownData":false,"type":"int"},
+     * "eventStreamName":"process1_77_process_stream"}
+     * @throws RegistryException
+     */
+    public JsonObject getKPIConfiguration(String processDefinitionId) throws RegistryException {
+        String resourcePath = AnalyticsPublisherConstants.REG_PATH_BPMN_ANALYTICS + processDefinitionId + "/"
+                + AnalyticsPublisherConstants.ANALYTICS_CONFIG_FILE_NAME;
+        try {
+            RegistryService registryService = BPMNAnalyticsHolder.getInstance().getRegistryService();
+            Registry configRegistry = registryService.getConfigSystemRegistry();
+
+            if (configRegistry.resourceExists(resourcePath)) {
+                Resource processRegistryResource = configRegistry.get(resourcePath);
+                String dasConfigDetailsJSONStr = new String((byte[]) processRegistryResource.getContent(),
+                        StandardCharsets.UTF_8);
+                return new JsonParser().parse(dasConfigDetailsJSONStr).getAsJsonObject();
+            } else {
```

Comment:
else is not necessary here. You can simply move return null; below "if" and remove "else " statement.